src/MIDebugEngine/AD7.Impl/AD7Engine.cs

```diff
@@ -70,7 +70,14 @@ static AD7Engine()
 
         public AD7Engine()
         {
-            Host.EnsureMainThreadInitialized();
+            try
+            {
+                Host.EnsureMainThreadInitialized();
+            }
+            catch
```

Comment:
This should really be inside of the implementation of Host.EnsureMainThreadInitialized. I tried fixing this again: https://github.com/Microsoft/MIEngine/pull/512. Can you give that a try?